File: app/src/main/java/com/stevecrossin/whatcanicook/entities/LogRecords.java
package com.stevecrossin.whatcanicook.entities;


import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

/**
 * LogRecords database - definition of data model as it will be saved and handled in the database schema
 */
@Entity(tableName = "logs")
public class LogRecords {

    @PrimaryKey(autoGenerate = true)
    private int log_id;

    @ColumnInfo(name = "error_log")
    private String logs;

    /**
     * Getter and setter methods for the database. Each method returns or sets the relevant field in the database
     */

    int getLog_id() {
        return log_id;
    }

    public String getLogs() {
        return logs;
    }

    void setLog_id(int log_id) {
        this.log_id = log_id;
    }

    public void setLogs(String logs) {
        this.logs = logs;
    }

    public LogRecords(String logs) {
        this.logs = logs;
    }
}

File: app/src/main/java/com/stevecrossin/whatcanicook/entities/LogRecords.kt
package com.stevecrossin.whatcanicook.entities

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

/**
 * LogRecords database - definition of data model as it will be saved and handled in the database schema
 */
@Entity(tableName = "logs")
class LogRecords(@field:ColumnInfo(name = "error_log")
                 var logs: String?) {

    /**
     * Getter and setter methods for the database. Each method returns or sets the relevant field in the database
     */

    @PrimaryKey(autoGenerate = true)
    internal var log_id: Int = 0
}
